barcodes_generator_product/views/view_product_product.xml

```diff
@@ -21,21 +21,23 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
                     'invisible': [('barcode_rule_id', '=', False)],
                     'readonly': [('generate_type', '!=', 'manual')]}"
                     groups="barcodes_generator_abstract.generate_barcode" colspan="2"/>
-                <button name="generate_base" type="object" string="Generate Base (Using Sequence)" attrs="{'invisible': ['|',
-                    ('generate_type', '!=', 'sequence'),
-                    ('barcode_base', '!=', 0)]}"
+                <button name="generate_base" type="object"
+                    string="Generate Barcode Base"
+                    help="Generate Barcode Base from the sequence configured on the Barcode Rule"
+                    attrs="{'invisible': ['|',
+                        ('generate_type', '!=', 'sequence'),
+                        ('barcode_base', '!=', 0)]}"
                     groups="barcodes_generator_abstract.generate_barcode" colspan="2"/>
             </field>
 
-            <field name="barcode" position="attributes">
-                <attribute name="attrs">{'readonly': [('generate_type', '=', 'sequence')]}</attribute>
-            </field>
-
             <field name="barcode" position="after">
-                <button name="generate_barcode" type="object" string="Generate Barcode (Using Barcode Rule)"
-                    attrs="{'invisible': ['|',
+                <button name="generate_barcode" type="object"
+                    string="Generate Barcode"
+                    help="Generate a barcode using the selected Barcode Rule and the value of the Barcode Base"
+                    attrs="{'invisible': ['|', '|',
                             ('barcode_rule_id', '=', False),
-                            ('barcode_base', '=', 0)]}"
+                            ('barcode_base', '=', 0),
+                            ('barcode', '!=', False)]}"
```

Comment:
Not convinced for this point.  (just the ('barcode', '!=', False) condition.

This will prevent users to regenerate barcode, that is usefull  if you change the pattern of your barcode, changing prefix from 021... to 022.... for exemple.

Reply:
Right, I didn't have this scenario in mind (not use if changing barcodes on existing product is a scenario that is supposed to happen, but...). In this case, they would have to empty the barcode field and then the button will be displayed again.

By the way, I forgot to say in the comments that I also removed the readonly attribute on the barcode field when generated from sequence so that users can remove a barcode from a product (this happens when they re-use an existing barcode for another product)

Reply:
OK. Well, the problem I see in this configuration is the possiblity to write a bad barcode associated to a product with barcode_rule_id defined.

A medium proposal could be : 
- hide button generate Barcode if barcode is defined. (what you already did)
- let barcode readonly (to secure data). (by the way, should be readonly if barcode_rule_id != False).
- Add an extra button "Clear barcode" that set barcode to False, displayed if barcode is defined and barcode_rule_id is defined.

what do you think about that proposal ? More secure, and allow to change barcodes.

Reply:
I find it a bit heavy... but I can understand that when you have "untrusted" users for which you want to lock down things and block mistakes as much as possible. So I just implemented it your way.

Reply:
Thanks a lot. :+1: